mex.go

```diff
@@ -186,6 +187,7 @@ func (mexset *messageExchangeSet) newExchange(ctx context.Context, framePool Fra
 
 	mexset.exchanges[mex.msgID] = mex
 	mexset.sendChRefs.Add(1)
+	mexset.onAdded()
```

Comment:
This is called with the mexset locked, which is not right. This could cause deadlocks, since an exchange update could cause a peer score to update, but the peer score needs the incoming count, which needs a read lock on the mexset which is already locked.

Unlock the `mexset` after the `mexset.exchanges` set, and before the return earlier.